diff --git a/betterdiscordctl b/betterdiscordctl
index 9582be4..cccea9f 100755
--- a/betterdiscordctl
+++ b/betterdiscordctl
@@ -126,6 +126,9 @@ while :; do
     --flavors=)
       die_non_empty '--flavors'
       ;;
+    --nix)
+      scan=("$(nix-store -r $(which Discord) 2>/dev/null)/opt")
+      ;;
     -d|--discord)
       if [[ ${2+x} ]]; then discord=$2; shift
       else die_non_empty '--discord'; fi
diff --git a/default.nix b/default.nix
new file mode 100644
index 0000000..3663520
